host/resetworkflow_test.go

```diff
@@ -200,3 +203,247 @@ func (s *integrationSuite) TestResetWorkflow() {
 	s.NotNil(firstActivityCompletionEvent)
 	s.True(workflowComplete)
 }
+
+func (s *integrationSuite) TestResetWorkflow_ReapplyAll() {
```

Comment:
Both these tests are nearly identical.  I highly recommend to refactor it in a way which allows us to reuse the logic from both integration tests rather than copy pasting it.

Reply:
it depends, my current preference is to make all test self contained, e.g. no test library, no test dependency, no test magic